### PR TITLE
Reworked and fixed dossier deactivation and activation

### PR DESCRIPTION
**Pull request by phgross**
Reworked and fixed dossier deactivation:
 - Add precondition checks for not checked_in documents (fixes Issue #689 )
 - Extend tests
 - Refactoring and cleanup (plone.api, splitting up ect.).

@lukasgraf @deiferni please have a look

### CODE REVIEW

**Comment by phgross** [comment]
@lukasgraf @deiferni I've extend my PR with a small rework of the dossier activation, please have a look ...

File: opengever/dossier/activate.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+from five import grok
+from opengever.dossier import _
+from opengever.dossier.behaviors.dossier import IDossierMarker
+from plone import api
+from Products.statusmessages.interfaces import IStatusMessage
+
+
+class DossierActivateView(grok.View):
+    """View wich activate the dossiers including his subdossiers."""
```
Comment (lukasgraf) [inline opengever/dossier/activate.py]: `wich activate the dossiers including his subdossiers` -> `wich activates the dossier including its subdossiers`

**Comment by lukasgraf** [comment]
:+1: - but changelog entry is missing.

**Comment by phgross** [comment]
updated and added changelog entry

File: opengever/dossier/deactivate.py [hunk 2]
```diff
@@ -15,57 +15,47 @@ class DossierDeactivateView(grok.View):
...
-                IStatusMessage(self.request).add(
-                    _("The Dossier can't be deactivated, the subdossier "
-                      "${dossier} is already resolved",
-                      mapping=dict(dossier=subdossier.Title.decode('utf-8'),)),
-                    type='error')
-                return self.request.RESPONSE.redirect(
-                    self.context.absolute_url())
-
-        # recursively resolve all dossier
+        # recursively deactivate all dossier
```
Comment (deiferni) [inline opengever/dossier/deactivate.py]: dossier -> dossiers

File: opengever/dossier/deactivate.py [hunk 3]
```diff
@@ -15,57 +15,47 @@ class DossierDeactivateView(grok.View):
...
-        # deactivate main dossier
-        wft.doActionFor(
-            self.context, 'dossier-transition-activate')
-
-        IStatusMessage(self.request).add(
-            _("The Dossier has been activated"), type='info')
+            state = api.content.get_state(obj=subdossier.getObject())
+            if state == 'dossier-state-resolved':
+                IStatusMessage(self.request).add(
+                    _("The Dossier can't be deactivated, the subdossier "
```
Comment (deiferni) [inline opengever/dossier/deactivate.py]: I think this should also be unicode

**Comment by deiferni** [comment]
looks :+1: apart form my two picky comments :wink:

**Comment by phgross** [comment]
@deiferni updated according to your comments and updated added also the german translations

**Comment by deiferni** [comment]
:tada:

**Pull request merged**